modules/gltf/src/gltf-writer.ts

```diff
@@ -26,18 +28,24 @@ export const GLTFWriter = {
     gltf: {}
   },
 
-  encode: async (gltf, options: GLTFWriterOptions = {}) => encodeSync(gltf, options),
+  encode: async (gltf: GLTFWithBuffers, options: GLTFWriterOptions = {}) =>
+    encodeSync(gltf, options),
   encodeSync
 } as WriterWithEncoder<any, never, GLTFWriterOptions>;
 
-function encodeSync(gltf, options: GLTFWriterOptions = {}) {
+function encodeSync(gltf: GLTFWithBuffers, options: GLTFWriterOptions = {}) {
   const {byteOffset = 0} = options;
 
+  encodeExtensions(gltf);
+  const scenegraph = new GLTFScenegraph(gltf);
+  scenegraph.createBinaryChunk();
+  const gltfToEncode = scenegraph.gltf;
```

Comment:
These changes are not tested. Add a test with extension to `gltf-writer.spec.ts`

Reply:
Done.